tests/unittest.cpp

```diff
@@ -1,1290 +1,1247 @@
-    CROW_ROUTE(app1, "/").methods("GET"_method, "POST"_method)([]{return "A";});
-    CROW_ROUTE(app2, "/").methods("GET"_method, "POST"_method)([]{return "B";});
-
-    //Server<SimpleApp> server1(&app1, LOCALHOST_ADDRESS, 45451);
-    //Server<SimpleApp> server2(&app2, LOCALHOST_ADDRESS, 45452);
-
-    //auto _ = async(launch::async, [&]{server1.run();});
-    //auto _2 = async(launch::async, [&]{server2.run();});
-    auto _ = async(launch::async, [&]{app1.bindaddr(LOCALHOST_ADDRESS).port(45451).run();});
-    auto _2 = async(launch::async, [&]{app2.bindaddr(LOCALHOST_ADDRESS).port(45452).run();});
-    app1.wait_for_server_start();
-    app2.wait_for_server_start();
-
-    std::string sendmsg = "POST /\r\nContent-Length:3\r\nX-HeaderTest: 123\r\n\r\nA=B\r\n";
-    asio::io_service is;
-    {
-        asio::ip::tcp::socket c(is);
-        c.connect(asio::ip::tcp::endpoint(asio::ip::address::from_string(LOCALHOST_ADDRESS), 45451));
-
-        c.send(asio::buffer(sendmsg));
-
-        size_t recved = c.receive(asio::buffer(buf, 2048));
-        ASSERT_EQUAL('A', buf[recved-1]);
+  static char buf[2048];
+  SimpleApp app1, app2;
+  CROW_ROUTE(app1, "/").methods("GET"_method,
+                                "POST"_method)([] { return "A"; });
+  CROW_ROUTE(app2, "/").methods("GET"_method,
+                                "POST"_method)([] { return "B"; });
+
+  auto _ = async(launch::async,
+                 [&] { app1.bindaddr(LOCALHOST_ADDRESS).port(45451).run(); });
+  auto _2 = async(launch::async,
+                  [&] { app2.bindaddr(LOCALHOST_ADDRESS).port(45452).run(); });
+  app1.wait_for_server_start();
+  app2.wait_for_server_start();
+
+  std::string sendmsg =
+      "POST /\r\nContent-Length:3\r\nX-HeaderTest: 123\r\n\r\nA=B\r\n";
+  asio::io_service is;
+  {
+    asio::ip::tcp::socket c(is);
+    c.connect(asio::ip::tcp::endpoint(
+        asio::ip::address::from_string(LOCALHOST_ADDRESS), 45451));
+
+    c.send(asio::buffer(sendmsg));
+
+    size_t recved = c.receive(asio::buffer(buf, 2048));
+    REQUIRE('A' == buf[recved - 1]);
+  }
+
+  {
+    asio::ip::tcp::socket c(is);
+    c.connect(asio::ip::tcp::endpoint(
+        asio::ip::address::from_string(LOCALHOST_ADDRESS), 45452));
+
+    for (auto ch : sendmsg) {
+      char buf[1] = {ch};
+      c.send(asio::buffer(buf));
     }
 
-    {
-        asio::ip::tcp::socket c(is);
-        c.connect(asio::ip::tcp::endpoint(asio::ip::address::from_string(LOCALHOST_ADDRESS), 45452));
-
-        for(auto ch:sendmsg)
-        {
-            char buf[1] = {ch};
-            c.send(asio::buffer(buf));
-        }
-
-        size_t recved = c.receive(asio::buffer(buf, 2048));
-        ASSERT_EQUAL('B', buf[recved-1]);
-    }
+    size_t recved = c.receive(asio::buffer(buf, 2048));
+    REQUIRE('B' == buf[recved - 1]);
+  }
 
-    app1.stop();
-    app2.stop();
+  app1.stop();
+  app2.stop();
 }
 
-TEST(json_read)
+TEST_CASE("json_read")
 {
-	{
-        const char* json_error_tests[] =
```

Comment:
there seems to be some sorting here, I'm guessing the formatter messed it up

Reply:
Looking at test logic, order of array's element shouldn't matter. And it looked like formatter only put one entry per line.

Reply:
yeah I know, I just thought it looked more organized earlier.